bankbot.lic

```diff
@@ -0,0 +1,229 @@
+    unless has_bot_account(character)
+      fput "whisper #{character} Apologies, but my record shows that you do not have a Bot Bank account."
+      pause 0.5
+      fput "whisper #{character} You can tip me to start one."
+      return
+    end
+
+    currency_type = get_currency_type(denomination)
+
+    balance = get_balance(character, currency_type)
+    if balance < amount.to_i
+      fput "whisper #{character} The max you can withdraw is: #{balance} #{currency_type}."
+      return
+    end
+
+    case bput("tip #{character} #{amount} kronar", '\w+ declines your tip offer.', 'Your tip offer to \w+ has expired.', '\w+ accepts your tip and slips it away with a smile')
+    when /\w+ declines your tip offer.|Your tip offer to \w+ has expired./i
+      fput "whisper #{character} Alright then, I will continue to hold on to it."
+    when /\w+ accepts your tip and slips it away with a smile/i
+      balance = balance - amount.to_i
+      if balance <= 0
+        fput "whisper #{character} Your #{currency_type} account has now been closed!"
+        UserVars.bot_bank[character][currency_type] = 0
+      else
+        fput "whisper #{character} You now have a remaining balance of: #{balance} #{currency_type}."
+        UserVars.bot_bank[character][currency_type] = balance
+
+        show_balance_sheet(character)
+      end
+    else
+      fput "whisper #{character} I was unable to process your withdrawal. Please try again later."
+      fput "cancel tip"
+    end
+  end
+
+  def ping
+    return if @last_action + 60 * 3 > Time.now
+    @last_action = Time.now
+  end
+
+  private
+
+  def get_balance(character, denomination)
+    unless has_bot_account(character)
+      return 0
+    else
+      echo ("Trying to return: #{UserVars.bot_bank[character][denomination]}") if @bankbot_debug
+      return UserVars.bot_bank[character][denomination].to_i
+    end
+  end
+
+  def has_bot_account(character)
+    unless UserVars.bot_bank.has_key?(character)
+      return false
+    else
+      return true
+    end
+  end
+
+  def get_currency_type(denomination)
+    case denomination
+    when /Dok|Dokora.*/i
+      currency_type = "Dokora"
+    when /Lir|Lirum.*/i
+      currency_type = "Lirum"
+    when /Kro|Kronar.*/i
+      currency_type = "Kronar"
+    end
+
+    return currency_type
+  end
+
+  def show_balance_sheet(character)
+    # Formatted balance listing
+
+    @denominations.each do |type|
+      account_balance = UserVars.bot_bank[character][type]
+
+      if account_balance <= 0 || account_balance.nil?
+        next
+      end
+
+      money_readable = minimize_coins(account_balance)
+
+      denomination_msg = ""
+      money_readable.each_with_index do |amount, index|
+        denomination_msg = denomination_msg + "#{money_readable[index]} " 
+      end
+
+      fput "whisper #{character} It looks like your current balance is #{denomination_msg} (#{account_balance}) #{type}."
+    end
+  end
+
+end
+
+arg_definitions = [
```

Comment:
There's no arg for `sleep`, yet it is referenced on line 180. I recommend changing line 180 to `false` to make it clear.

Although, to be honest, I've been considering removing that functionality from lichbot since it never gets used...